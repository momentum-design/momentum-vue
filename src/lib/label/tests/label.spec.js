import { shallowMount } from '@vue/test-utils';
import Label from '../index.vue';

describe('Label', () => {
  it('should match snapshot', () => {
    const wrapper = shallowMount(Label, {
      propsData: {
        htmlFor: 'test',
        label: 'testLabel'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should have HTML for attribute', () => {
    const wrapper = shallowMount(Label, {
      propsData: {
        htmlFor: 'test',
        label: 'testLabel'
      }
    });

    expect(wrapper.find('label').attributes().for).toEqual('test');
  });

  it('should render label', () => {
    const wrapper = shallowMount(Label, {
      propsData: {
        htmlFor: 'test',
        label: 'testLabel'
      }
    });

    expect(wrapper.find('span').text()).toEqual('testLabel');
  });

  it('should apply dark theme class', () => {
    const wrapper = shallowMount(Label, {
      propsData: {
        htmlFor: 'test',
        label: 'testLabel',
        theme: 'dark'
      }
    });

    expect(wrapper.classes('md-label--dark')).toBe(true);
  });
});
