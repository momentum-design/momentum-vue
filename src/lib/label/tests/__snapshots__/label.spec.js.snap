// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should match snapshot 1`] = `<label for="test" class=""><span>testLabel</span></label>`;
